src/Meltor.php

```diff
@@ -0,0 +1,872 @@
+            'Dump completed on (removed for comparison)',
+            $sqlStructure
+        );
+    }
+
+    /**
+     * Reads a given database structure file for comparison, and removes incomparable noise from the content.
+     *
+     * The resulting content is only suitable for comparison.
+     *
+     * @param string $structureFileName
+     *
+     * @return string
+     */
+    public function readAndCleanStructure(string $structureFileName): string
+    {
+        $structure = file_get_contents($structureFileName);
+        $structure = $this->removeTimeStampFromDump($structure);
+        $structure = $this->removeRedundantCharSetFromDump($structure);
+
+        return $structure;
+    }
+
+    /**
+     * Writes a new migration into the "migrations" folder.
+     *
+     * Older migrations with the same name will be removed
+     * - in order to keep a current file name to run last
+     * - while not filling the "migrations" folder with each run
+     *
+     * Will return the name of the new migration file.
+     *
+     * @param string $migrationContent
+     * @param string $migrationName
+     * @param int $offset Seconds to add to the file name date
+     * @param Command|null $command Used for shell output
+     *
+     * @return string
+     */
+    public function writeMigration(string $migrationContent, string $migrationName, Command $command = null, int $offset = 0): string
+    {
+        $filename          = sprintf('%s_%s.php', now()->addSeconds($offset)->format('Y_m_d_His'), $migrationName);
+        $migrationFilePath = $this->combinePath($this->config('migration.folder'), $filename);
+
+        $this->removeMigrations($migrationName, $command);
+        $command?->info(sprintf('Generate new migration "%s"', $migrationFilePath));
+        file_put_contents($migrationFilePath, $migrationContent);
+
+        return $migrationFilePath;
+    }
+
+    /**
+     * Removes previously created migrations of a given name.
+     *
+     * @param string $migrationName
+     * @param Command|null $command Used for shell output
+     * @return void
+     */
+    public function removeMigrations(string $migrationName, Command $command = null): void
+    {
+        $migrationFolder     = $this->config('migration.folder');
+        $oldMigrationPattern = sprintf('%s/????_??_??_??????_%s.php', $migrationFolder, $migrationName);
+
+        foreach (File::glob($oldMigrationPattern) as $oldMeltorMigration) {
+            $command?->warn(sprintf('Removing old migration "%s"', $oldMeltorMigration));
+            File::delete($oldMeltorMigration);
+        }
+    }
+
+    /**
+     * Restore the database to its prior state.
+     *
+     * To be used if something goes wrong during a testrun, where the database is being modified for comparison.
+     *
+     * @param Command|null $command Used for shell output
+     * @return void
+     */
+    public function restoreBackup(Command $command = null): void
```

Comment:
This class has a lot of very different responsibilities. Maybe you should consider splitting this into multiple classes.